### DISCUSSION TITLE
List all tiles in viewport

### DISCUSSION THREAD

**Discussion by jo-chemla** [uber/react-map-gl#2247] [Q&A] [answered]
Following a question on the mapbox-gl-js repo [here](https://github.com/mapbox/mapbox-gl-js/issues/12826), I wondered if this is a feature which is offered by react-map-gl itself - I've got a feeling I've seen something similar around the react-map-gl github repo or doc in the past. 

I've got a map with a TMS source and layer, and I'd like to get a list of all tiles displayed for current viewport. This would allow me to parse the tiles URLs and fetch some header propreties of the responses of all the tile requests (to aggregate a tile-date per-tile property into a min/max date for the viewport). 

Thanks a lot for your help - and again, really useful library. 

Edit: what I'm after is probably inside [math.gl](https://uber-web.github.io/math.gl/docs). Getting visibleTiles should be an easy task - although I would have hoped to be able to get it directly from a loadedTiles list to avoid discrepancies between the tiles loaded in the viewport and the computation done via another piece of code. 

Plus an existing solution would let me get loaded tiles for all configurations (globe or equal-earth projection, pitch different than zero etc rather than only for the standard web-mercator case).

**Comment by Pessimistress** [comment, marked as answer]
react-map-gl is just a wrapper of the base map library. It does not contain any tile fetching or rendering logic. If mapbox-gl adds an API you can access it via the MapRef object.

**Reply by jo-chemla** [reply]
Thanks for getting back so promptly. Indeed, `listing all tiles rendered in viewport` does not seem like a feature included in the mapbox api at the moment. Maybe this question could have been asked directly on the math-gl repo, although this would mean computing the list of tiles displayed in parallel of loading them via the mapping lib, hence might introduce discrepancies. Will check on the mapbox-gl thread to see if there is a more direct way, thanks again!
